### PR TITLE
Update state to handle named unit creation and embedded charms.

### PR DESCRIPTION
**Pull request by hpidcock**
## Description of change

Update state to handle named unit creation and embedded charms.

## QA steps

Unit Tests

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: state/unit.go [hunk 1]
```diff
@@ -132,6 +132,27 @@ func (u *Unit) ShouldBeAssigned() bool {
 	return u.modelType == ModelTypeIAAS
 }
 
+// IsEmbedded
```
Comment (ycliuhw) [inline state/unit.go:135]: better comment?

**Review by ycliuhw — APPROVED** [review]
lgtm thanks

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**